src/components/DataFetcher.tsx

```diff
@@ -21,13 +21,14 @@ const DataFetcher = <E extends unknown>({
   dataFetcher,
   children,
 }: DataFetcherProps<E>) => {
-  const [state, send] = useMachine(
-    createDataFetcherMachine({
+  const dataFetcherMachine = React.useMemo(() => {
```

Comment:
Is the general idea that useMemo stops the machine from constantly running?

Reply:
The warning was "Machine given to `useMachine` has changed between renders. This is not supported and might lead to unexpected results.
Please make sure that you pass the same Machine as argument each time."
This prevents from giving a different instance of machine to useMachine on each render unless there is a change in required parameters. Please refer [https://xstate.js.org/docs/tutorials/reddit.html#splitting-machines](https://xstate.js.org/docs/tutorials/reddit.html#splitting-machines)